habitat-lab/habitat/sims/habitat_simulator/sim_utilities.py

```diff
@@ -824,6 +964,58 @@ def ontop(
     return ontop_object_ids
 
 
+def on_floor(
+    sim: habitat_sim.Simulator,
+    object_a: habitat_sim.physics.ManagedRigidObject,
+    distance_threshold: float = 0.04,
+    alt_pathfinder: habitat_sim.nav.PathFinder = None,
+    island_index: int = -1,
+    ao_link_map: Dict[int, int] = None,
+    ao_aabbs: Dict[int, mn.Range3D] = None,
+) -> bool:
+    """
+    Gets whether or not the object is on the "floor" using the navmesh as an abstraction.
+    NOTE: alt_pathfinder option can be used to provide an alternative navmesh sized for objects. This would allow objects to be, for example, under tables or in corners and still be considered on the navmesh.
+
+    :param sim: The Simulator instance.
+    :param object_a: The object instance.
+    :param distance_threshold: Maximum allow-able displacement between current object position and navmesh snapped position.
+    :param alt_pathfinder:Optionally provide an alternative PathFinder specifically configured for this check. Defaults to sim.pathfinder.
+    :param island_index: Optionally limit allowed navmesh to a specific island. Default (-1) is full navmesh.
+    :param ao_link_map: A pre-computed map from link object ids to their parent ArticulatedObject's object id.
+    :param ao_aabbs: A pre-computed map from ArticulatedObject object_ids to their local bounding boxes. If not provided, recomputed as necessary.
+
+    :return: Whether or not the object is considered "on the floor" given the configuration.
+    """
+
+    assert isinstance(
+        object_a, habitat_sim.physics.ManagedRigidObject
+    ), "Object must be ManagedRigidObject, not implemented for ArticulatedObjects or links."
```

Comment:
Just checking -- do we have use cases for AO on_floor?

Reply:
I don't think so because we don't support AO "pick-able" objects presently. I currently assume all AOs are furniture or robot/human.